src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+        }
+        if (repo2.IsLocal)
+        {
+            var res = await processManager.ExecuteGit(repo2.Path, "status");
+            res.ThrowIfFailed($"{repo2.Path} is not a git repo");
+        }
+    }
+        
+    private bool IsLocalRepoVmr(string repoPath) => fileSystem.FileExists(Path.Combine(repoPath, VmrInfo.SourceDirName, VmrInfo.SourceMappingsFileName));
+    private async Task<bool> IsRemoteRepoVmr(string uri, string branch)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(uri);
+        try
+        {
+            await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", uri, branch);
+            return true;
+        }
+        catch(Exception)
+        {
+            return false;
+        }
+    }
+
+    private record DiffRepo(string Path, string Branch, bool IsLocal, bool IsVmr);
+
+    private async Task AddRemoteAndGenerateDiff(string repo1, string repo2, string repo2Branch, IReadOnlyCollection<string> filters)
+    {
+        string remoteName = Guid.NewGuid().ToString();
+
+        await processManager.ExecuteGit(repo1, [
+                "remote", "add", remoteName, repo2
+            ]);
+        var res = await processManager.ExecuteGit(repo1, [
+                "fetch", remoteName,
+                "--depth=1"
+            ]);
+        var sha1 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", "HEAD"
+            ])).StandardOutput.Trim();
+        var sha2 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", $"{remoteName}/{repo2Branch}"
+            ])).StandardOutput.Trim();
+
+        var patches = await patchHandler.CreatePatches(
+            options.OutputPath,
+            sha1,
+            sha2,
+            path: null,
+            filters,
+            relativePaths: false,
+            workingDir: new NativePath(repo1),
+            applicationPath: null,
+            CancellationToken.None);
+
+        if (patches.Count == 1)
+        {
+            using FileStream fs = new(patches[0].Path, FileMode.Open, FileAccess.Read);
+            using StreamReader sr = new(fs);
+            string? line;
+            while ((line = await sr.ReadLineAsync()) != null)
+            {
+                Console.WriteLine(line);
+            }
+        }
+        else
+        {
+            Console.WriteLine("Patch was too big so it had to be split into multiple files");
+            if (UxHelpers.PromptForYesNo("Do you want to apply the patches without seeing them?"))
+            {
+                foreach (var patch in patches)
+                {
+                    await patchHandler.ApplyPatch(patch, new NativePath(repo1), removePatchAfter: true);
+                }
+            }
+            else
+            {
+                foreach (var patch in patches)
+                {
+                    Console.WriteLine($"Patch file: {patch.Path} should be applied to {patch.ApplicationPath}");
```

Comment:
It would be better if we gave people the `git apply` commands instead of this text